sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriterContainer.scala

```diff
@@ -324,6 +324,8 @@ private[sql] class DynamicPartitionWriterContainer(
 
   private def bucketIdExpression: Option[Expression] = for {
     BucketSpec(numBuckets, _, _) <- bucketSpec
+  // Use `Pmod` to match the behaviour of `Utils.nonNegativeMod`, which is the function used to
```

Comment:
I don't think this is a good place for this comment. It needs to be with Pmod and Utils.nonNegativeMod indicating they can't be changed. (Can we also implement Pmod as just calling Utils.nonNegativeMod?) That will make sure they stay in sync.